smsdk/client.py

```diff
@@ -277,4 +259,31 @@ def get_type_from_machine(self, machine_source=None, **kwargs):
             self.config["protocol"], self.tenant, self.config["site.domain"]
         )
         return machine(self.session, base_url).get_type_from_machine_name(machine_source, **kwargs)
+    
+    def get_machine_schema(self, machine_source, types=[], show_hidden=False, return_mtype=False, **kwargs):
+        machineType= smsdkentities.get('machine_type')
+        machine_type = self.get_type_from_machine(machine_source)
```

Comment:
The decorator also allowed users pass the `source_clean` instead of the `source`. Is there any reason we can't just continue to use the decorator?

Reply:
The decorator is grabbing all the info for the function and is using v0 routes to do so.  I would basically have to rewrite entirely anyways.

Reply:
The problem is: we can't change the SDK behavior in a way that could break existing usage. You are free to solve that however you want: e.g. rewrite the decorator to use v1 APIs, move its functionality to a separate util/decorator, or something else.